### ISSUE TITLE
Wiki Example Code Seems Outdated

### ISSUE THREAD

**Issue by DarioS** [cafferychen777/MicrobiomeStat#55]
In [Feature-level Analysis](https://www.microbiomestat.wiki/single-point-analysis/cross-sectional-investigation-taxa-analysis-with-microbiomestat), I see `generate_taxa_volcano_single` example but function is not present in version 1.2 on CRAN.

```r
> library(MicrobiomeStat)
> generate_taxa_volcano_single
Error: object 'generate_taxa_volcano_single' not found
```

Also, is there plan to add [MA plot](https://en.wikipedia.org/wiki/MA_plot) to package? It would be a nice addiiton.

**Comment by cafferychen777** [comment]
Dear Dario,

Thank you for reaching out regarding the `generate_taxa_volcano_single` function and the MA plot feature in the MicrobiomeStat package.

Please note that the MicrobiomeStat package is currently under active development, and many of the latest features have not yet been released to CRAN. To access the most up-to-date functionality, including the `generate_taxa_volcano_single` function, we recommend using the GitHub version of the package.

For detailed instructions on setting up the GitHub version of MicrobiomeStat, please refer to our installation guide: "https://www.microbiomestat.wiki/setting-up-microbiomestat-installation-and-data-preparation/kick-start-your-journey-the-microbiomestat-installation-guide"

Regarding the MA plot, we appreciate your suggestion and will consider adding this feature in a future release of the package.

If you have any further questions or concerns, please don't hesitate to reach out.

Best regards,
Chen

**Comment by DarioS** [comment]
It is confusing because main branch is version 1.1.3. So, I thought CRAN is newer than GitHub because 1.2 is larger than 1.1.3.

**Comment by cafferychen777** [comment]
Hi Dario,

I apologize for the confusion caused by the inconsistent versioning between the CRAN and GitHub releases of MicrobiomeStat.

You're correct that the main branch on GitHub is labeled as version 1.1.3, while the CRAN version is 1.2. However, the GitHub version actually contains more recent features and updates than the CRAN release, despite having an older version number.

This discrepancy is an oversight on my part. I should have updated the version number on GitHub to reflect the latest changes. I will rectify this as soon as possible to ensure clarity and consistency across the different release channels.

To summarize:
- The GitHub version, although labeled 1.1.3, is the most up-to-date and includes features like `generate_taxa_volcano_single` that are not yet available in the CRAN release.
- I will update the GitHub version number to properly reflect its current state and avoid further confusion.

Thank you for bringing this issue to my attention. I appreciate your understanding and patience as I work to resolve the versioning inconsistency.

Best regards,
Caffery

**Comment by cafferychen777** [comment]
Dear Dario,

Great news! I'm pleased to inform you that we have just added the MA plot functionality to the MicrobiomeStat package, as you suggested. The new function `generate_taxa_ma_plot_single` has been implemented and is now available in the latest GitHub version of the package.

This function creates MA plots to visualize differential abundance analysis results, displaying the log2 fold change (M) against the average abundance (A) for each taxon. It's particularly useful for identifying taxa with significant abundance differences between groups while accounting for their baseline abundance levels.

Here's a simple example of how to use the new function:

```r
# Load the package (GitHub version)
library(MicrobiomeStat)

# Run differential abundance analysis
test.list <- generate_taxa_test_single(
  data.obj = ecam.obj,
  group.var = "delivery",
  feature.level = c("Genus")
)

# Generate MA plot
plot.list <- generate_taxa_ma_plot_single(
  data.obj = ecam.obj,
  group.var = "delivery",
  test.list = test.list,
  feature.sig.level = 0.05,
  feature.mt.method = "fdr"
)

# View the plot
print(plot.list$Genus[[1]])
```

The function includes various customization options, such as adjusting significance levels, multiple testing correction methods, and color schemes.

Thank you for your valuable suggestion that helped improve the MicrobiomeStat package. We're committed to continuously enhancing the package based on user feedback and needs.

Best regards,
Chen

**Comment by cafferychen777** [comment]
This documentation issue has been resolved, and the requested MA plot feature has been implemented.

The `generate_taxa_ma_plot_single()` function is now available in the latest GitHub version (as mentioned in the previous comment from April 2025). The Wiki documentation has been updated to reflect current functionality.

Install the latest version to access all features:
```r
devtools::install_github('cafferychen777/MicrobiomeStat')
```

Closing this issue as both the documentation concern and the MA plot feature request have been addressed. Feel free to reopen if you encounter any related issues with the latest version.
